Client/Pages/Transfer.razor

```diff
@@ -0,0 +1,155 @@
+                    }
+                    else
+                    {
+                        <small class="form-text text-danger">Recipient's name is invalid.</small>
+                    }
+                </div>
+                <div class="col-6">
+                    <label>Recipient's Wallet Id :</label>
+                    <input class="form-control" placeholder="Recipient's Wallet Id..." @bind="targetId">
+                    @if (StatusData(2,targetId))
+                    {
+                        <small class="form-text text-success">The Wallet Id format is valid.</small>
+                    }
+                    else
+                    {
+                        <small class="form-text text-danger">The Wallet Id format is invalid.</small>
+                    }
+                </div>
+            </div>
+            <div class="row">
+                <div class="col-6 p-2">
+                    <label>Amount :</label>
+                    <input type="text" class="form-control" placeholder="The Amount..." @bind="Amount">
+                    @if (StatusData(3,Amount))
+                    {
+                        <small class="form-text text-success">The amount entered is valid.</small>
+                    }
+                    else
+                    {
+                        <small class="form-text text-danger">The amount entered is invalid.</small>
+                    }
+                </div>
+                <div class="col-6 p-3">
+                    <br />
+                    @if (StatusData(1,data.Username) && StatusData(2,targetId) && StatusData(3,Amount))
+                    {
+                        <button class="btn btn-danger" @onclick=MakeTransfer>Finish the transfer</button>
+                        <small class="form-text text-success">The transfer can be done now.</small>
+                    }
+                    else
+                    {
+                        <button class="btn btn-danger">Finish the transfer</button>
+                        <small class="form-text text-danger">The transfer can't be done now.</small>
+                    }
+                </div>
+            </div>
+
+        </div>
+    </div>
+    <br />
+    @if (!Status)
+    {
+        <div class="alert alert-danger" role="alert">
+            Atentie ! Datele introduse nu corespund aceluiasi utilizator !
+        </div>
+    }
+}
+@code {
+    [Parameter]
+    public string sourceWalletId { get; set; }
+    public Wallet wallet { get; set; }
+
+    public List<string> users = new List<string>();
+    public bool Status = true;
+    public string Amount, targetId;
+    public TransferDto data = new TransferDto {
+        Username = null
+    };
+
+    protected override async Task OnInitializedAsync()
+    {
+        try
+        {
+            wallet = await HttpClient.GetFromJsonAsync<Wallet>($"api/wallet/{sourceWalletId}");
+            users = await HttpClient.GetFromJsonAsync<List<string>>($"api/helper");
```

Comment:
Imagine we have a million users. How well would this work?

Reply:
transfer.razor
private async Task CheckUsername()
    {
        usernameStatus = await HttpClient.GetFromJsonAsync<bool>($"api/helper/{data.Username}");
    }

controler
[HttpGet]
        [Route("{username}")]
        public bool CheckUser(string username)
        {
            foreach(var user in context.Users)
            {
                if (user.UserName == username)
                    return true;
            }
            return false;
        }`